### PR TITLE
Fix RocM resource class allocation

### PR DESCRIPTION
**Pull request by malfet**
Add Conf.is_test_stage() method to avoid duplicating state in ['test', 'test1', 'test2'] throughout the code

Test Plan: Make sure that in modified config.yml ROCM tests jobs are assigned `pytorch/amd-gpu` resource class

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -5856,12 +5858,14 @@ workflows:
...
           build_environment: "pytorch-linux-xenial-py3-clang5-asan-test1"
           docker_image: "308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-asan:ab1632df-fa59-40e6-8c23-98e004f61148"
+          resource_class: large
       - pytorch_linux_test:
           name: pytorch_linux_xenial_py3_clang5_asan_test2
           requires:
             - pytorch_linux_xenial_py3_clang5_asan_build
           build_environment: "pytorch-linux-xenial-py3-clang5-asan-test2"
           docker_image: "308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-xenial-py3-clang5-asan:ab1632df-fa59-40e6-8c23-98e004f61148"
+          resource_class: large
```
Comment (mrshenli) [inline .circleci/config.yml:5868]: Are these two changes relevant to this PR?

**Review by mrshenli — APPROVED** [review]
Stamp to unblock.

File: .circleci/config.yml (hunk 1 above)
Comment (malfet) [reply .circleci/config.yml:5868]: Yes, by default we run all tests on large resource class, but I might modify this one to keep `asan` tests running on medium (same CPU perf but less memory)
